### PR TITLE
bpo-45019: Cleanup module freezing and deepfreeze

### PR DESCRIPTION
**Pull request by kumaraditya303**
https://bugs.python.org/issue45019

### CODE REVIEW

**Comment by kumaraditya303** [comment]
cc @gvanrossum

**Comment by arhadthedev** [comment]
If a directory is renamed anyway, maybe `deepfrozen_modules` is better? `deepfreeze_modules` looks like "modules that are part of deepfreeze tool itself". Also it rhymes with `frozen_modules`.

**Review by gvanrossum — COMMENTED** [review]
Without looking carefully yet, I think I like your cleanup of freeze_modules.py, but not the rename of the deepfreeze directory. Can you undo the latter and just focus on improving the script?

**Comment by kumaraditya303** [comment]
> Without looking carefully yet, I think I like your cleanup of freeze_modules.py, but not the rename of the deepfreeze directory. Can you undo the latter and just focus on improving the script?

Sure, I will revert the directory name change tomorrow.

File: Tools/scripts/freeze_modules.py [hunk 1]
```diff
@@ -225,15 +223,16 @@ def iter_subs():
 #######################################
 # frozen source files
 
-class FrozenSource(namedtuple('FrozenSource', 'id pyfile frozenfile')):
+class FrozenSource(namedtuple('FrozenSource', 'id pyfile frozenfile deepfreezefile')):
 
     @classmethod
-    def from_id(cls, frozenid, pyfile=None, destdir=MODULES_DIR):
+    def from_id(cls, frozenid, pyfile=None, destdir=FROZEN_MODULES_DIR):
```
Comment (gvanrossum) [inline Tools/scripts/freeze_modules.py]: I wonder if we shouldn't just drop the destdir parameter, and hardcode FROZEN_MODULES_DIR where it's used. It's just passed around. (parse_frozen_specs() can also drop it then.) And things are asymmetrical -- you can override FROZEN_MODULES_DIR but not DEEPFROZEN_...

File: Tools/scripts/freeze_modules.py [hunk 2]
```diff
@@ -261,7 +260,7 @@ def ispkg(self):
             return os.path.basename(self.pyfile) == '__init__.py'
 
 
-def resolve_frozen_file(frozenid, destdir=MODULES_DIR):
+def resolve_frozen_file(frozenid, destdir=FROZEN_MODULES_DIR):
```
Comment (gvanrossum) [inline Tools/scripts/freeze_modules.py]: For this function, I'd just drop the default -- a value is always passed so nothing is lost.

**Review by gvanrossum — COMMENTED** [review]
This is great. I have a few nits.

I also wonder if there aren't more improvements we can make, e.g. use relative paths throughout the code, rather than computing absolute paths and then making them relative for the generated code again.

Oh, and it would be nice to be able to freeze *some* submodules in a package but not all of them -- e.g. I'd like to freeze `encodings/{__init__,utf_8,aliases}.py` but not the remaining 100 submodules in that package.

**Comment by kumaraditya303** [comment]
> This is great. I have a few nits.
> 
> I also wonder if there aren't more improvements we can make, e.g. use relative paths throughout the code, rather than computing absolute paths and then making them relative for the generated code again.
> 
> Oh, and it would be nice to be able to freeze *some* submodules in a package but not all of them -- e.g. I'd like to freeze `encodings/{__init__,utf_8,aliases}.py` but not the remaining 100 submodules in that package.

For improvements, this module has many small functions which make it hard to follow I would like to write a class and compress it to some methods to make it easy to follow, and Btw do you want type annotations as in deepfreeze.py for this module too ?

**Comment by kumaraditya303** [comment]
Regarding freezing encoding package I would do that in a separate PR to reduce this PR.

File: Tools/scripts/freeze_modules.py (hunk 1 above)
Comment (kumaraditya303) [reply Tools/scripts/freeze_modules.py]: Done

File: Tools/scripts/freeze_modules.py (hunk 2 above)
Comment (kumaraditya303) [reply Tools/scripts/freeze_modules.py]: Done

**Review by gvanrossum — APPROVED** [review]
Looks good for this phase! I'll merge this now. Then you can work on e.g. changes groups of functions to classes and partially freezing encodings.

**Comment by kumaraditya303** [comment]
Thanks @gvanrossum This was my first contribution to cpython earlier was just a typo fix, glad that you merged it :).

**Comment by gvanrossum** [comment]
Thanks Kumar for cleaning up our mess! :-) I am looking forward to seeing more from you.
